pytorch_lightning/trainer/training_io.py

```diff
@@ -119,6 +120,12 @@
 else:
     HOROVOD_AVAILABLE = True
 
+PRIMITIVE_TYPES = (
```

Comment:
@williamFalcon @awaelchli @yukw777 @festeh any other primitives shall be stored in checkpoint?

Reply:
I would generalize to Mapping, Sequence, etc, unless there is a reason why only support these specific types? :)

Reply:
I'm using dataclasses. They are actually great in the way that they give you auto-completion (plain config classes would allow it too) and can provide some validation. But I never heard that somebody else also doing that, so I'd probably could just patch this variable in my code.

In general I think we cannot handle all cases here so it would be beneficial to allow user to manually save some picklable argument, maybe via @should_pickle(argument) decorator. I'll try to design this feature.

Reply:
that sounds interesting, I would just stay for this PR to get done with the complete list, and kindly ask you to make a follow-up PR with your suggestion?

Reply:
yeah, sure